MetaMask/vault-decryptor
# Fix decoding of SRPs in vault data

We recently made a change in the extension to how keyring data is
encrypted so that the SRP is encoded. This means that when decrypting
the vault data the SRP must be decoded as well.

Fixes #27.
Fixes #32.
Fixes #36.
Fixes #4.

## Manual testing

1. Run `npm install`.
2. Run `npm run build` to create a build.
3. Run `open index.html` (macOS) or just open `index.html` in your browser.
4. Assuming you have the extension installed locally and you've had the extension installed for some time, copy your vault data as per the [guide in Zendesk](https://metamask.zendesk.com/hc/en-us/articles/360018766351-How-to-use-the-Vault-Decryptor-with-the-MetaMask-Vault-Data).
6. Paste it into the text area in the vault decryptor.
7. You should see the decrypted vault data appear along with your SRP.
8. Remove the extension from your browser.
9. Reinstall the extension and import your existing account or create a new one.
10. Copy your vault data again as per the guide above.
11. Paste it into the text area in the vault decryptor.
12. You should see the decrypted vault data appear along with your SRP.
